src/hooks/data/vaults/useVault.ts

```diff
@@ -1,41 +1,89 @@
-import { Address, useContractReads } from "wagmi"
+import { Address } from "wagmi"
 
+import {
+  findApiCompounderVaultForAsset,
+  findApiTokenVaultForAsset,
+} from "@/lib/findApiVaultForAsset"
 import { VaultProps } from "@/lib/types"
+import { useApiCompounderVaults, useApiTokenVaults } from "@/hooks/api"
 import useActiveChainId from "@/hooks/useActiveChainId"
+import { useFallbackReads } from "@/hooks/util"
 
 import { vaultCompounderAbi } from "@/constant/abi"
 
-// TODO: Use API first
-
 export function useVault({ asset, type, vaultAddress }: VaultProps) {
+  const apiCompoundersRequest = useApiCompounderVaults({ type })
+
   const isToken = type === "token"
   const chainId = useActiveChainId()
   const contract = { chainId, address: vaultAddress, abi: vaultCompounderAbi }
-  const vaultData = useContractReads({
-    contracts: [
-      { ...contract, functionName: "name" },
-      { ...contract, functionName: "symbol" },
-      { ...contract, functionName: "decimals" },
-      { ...contract, functionName: "totalAssets" },
-      { ...contract, functionName: isToken ? "name" : "getUnderlyingAssets" },
-    ],
-    enabled: !!vaultAddress,
-  })
-
-  const underlyingAssets = (
-    isToken || !vaultData.data ? [asset] : vaultData.data[4]
-  ) as Address[] | undefined
 
+  const apiCompounderVault = useApiCompounderVault({ asset, type })
+  const apiTokenVault = useApiTokenVault({ asset, type })
+
+  const fallbackRequest = useFallbackReads(
+    {
+      contracts: [
+        { ...contract, functionName: "name" },
+        { ...contract, functionName: "symbol" },
+        { ...contract, functionName: "decimals" },
+        {
+          ...contract,
+          functionName: isToken ? "name" : "getUnderlyingAssets",
+        },
+      ],
+      enabled: !!asset && !!vaultAddress,
+      select: (data) => ({
+        name: data[0],
+        symbol: data[1],
+        decimals: data[2],
+        underlyingAssets: isToken
+          ? ([asset] as Address[])
+          : (data[3] as Address[]),
+      }),
+    },
+    [apiCompoundersRequest, apiTokenVault]
+  )
+
+  return apiCompounderVault.isEnabled && !apiCompounderVault.isError
```

Comment:
I'd like to create a hook that returns the API/fallback query for us automatically so we don't need to repeat this constantly -- ran into some roadblocks so I'm going to continue like this for this PR